### PR TITLE
Feature/notify events

### PR DESCRIPTION
**Pull request by tomerdayan168**

### CODE REVIEW

File: components/events-notify.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import EventsNotifyView from "./events-notify-view";
+import { useEffect, useState } from "react";
+import {PIPE_API} from "../config"
```
Comment (Elon-Abulafia) [inline components/events-notify.js]:
```suggestion
import { PIPE_API } from "../config"
```

File: components/events-notify.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+function escapeRegExp(string) {
+  return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"); // $& means the whole matched string
+}
+
+function replaceAll(str, find, replace) {
+  return str.replace(new RegExp(escapeRegExp(find), "g"), replace);
+}
+
+async function notifyEvent(routine, event, args) {
+//   let SERVER_URL = "http://localhost:4000";
```
Comment (Elon-Abulafia) [inline components/events-notify.js]: Remove this

File: components/events-notify.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  let request = `${PIPE_API}/execute?event_name=${event}`;
+
+  if (routine !== null && routine !== undefined && routine.length === 2) {
+    request += `&specific_flow_routines={"${routine[0]}": ["${routine[1]}"]}`;
+  }
```
Comment (Elon-Abulafia) [inline components/events-notify.js]: Add a comment explaining this maybe?

File: components/events-notify.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  const [components, setComponents] = useState([]);
+
+  useEffect(() => {
+    fetch(`/api/events/`)
+      .then((res) => res.json())
+      .then((data) => {
+        let eventsTemp = [];
+        for (let event of data) {
+          eventsTemp.push(event["event_name"]);
+        }
```
Comment (San-Moshe) [inline components/events-notify.js]: I think that you should consider using `map` instead of `for loop`

File: components/events-notify.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        let componentsTemp = [];
+        for (let component of data) {
+          const fullName = component["full_name"];
+
+          // Remove the logger name from the full name
+          let loggerFlowRoutine = fullName.split(".");
+          loggerFlowRoutine.shift();
+
+          componentsTemp.push(loggerFlowRoutine);
+        }
```
Comment (San-Moshe) [inline components/events-notify.js]:
```suggestion
        components.map(component => component["full_name"].split(".").shift())
```

File: components/events-notify.js (hunk 3 above)
Comment (San-Moshe) [reply components/events-notify.js]: +1

File: components/events-notify.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  if (routine !== null && routine !== undefined && routine.length === 2) {
+    request += `&specific_flow_routines={"${routine[0]}": ["${routine[1]}"]}`;
+  }
+
+  if (args !== null) {
+    request += `&${replaceAll(args, ",", "&")}`;
+  }
+
+  try {
+    await fetch(request);
```
Comment (San-Moshe) [inline components/events-notify.js]: Should be `post` request instead of `get`.

File: components/events-notify.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+function replaceAll(str, find, replace) {
+  return str.replace(new RegExp(escapeRegExp(find), "g"), replace);
+}
+
+async function notifyEvent(routine, event, args) {
+//   let SERVER_URL = "http://localhost:4000";
+  let request = `${PIPE_API}/execute?event_name=${event}`;
+
+  if (routine !== null && routine !== undefined && routine.length === 2) {
+    request += `&specific_flow_routines={"${routine[0]}": ["${routine[1]}"]}`;
```
Comment (San-Moshe) [inline components/events-notify.js]: Should be inside request body (as json for example)

Comment (San-Moshe) [reply components/events-notify.js]: A better approach in my opinion includes some changes in the pipe API.
`PIPE_API/flows/routines/{routine_id}/events/{event_id}/execute`

And to include any additional arguments in req body as json.

Comment (San-Moshe) [reply components/events-notify.js]: I'll find some good articles on the topic.

**Review by San-Moshe — CHANGES_REQUESTED** [review]

File: components/events-notify.js (hunk 7 above)
Comment (tomerdayan168) [reply components/events-notify.js]: Sending a specific routine is optional. Maybe putting it in the body is a better solution.

File: components/events-notify.js (hunk 3 above)
Comment (tomerdayan168) [reply components/events-notify.js]: No comments. Extra line of code will do the trick to simplify that line.

**Review by Elon-Abulafia — APPROVED** [review]

**Pull request merged**